src/internal.c

```diff
@@ -12928,7 +12928,20 @@ static int DoHandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
         && ssl->error != WC_PENDING_E && ssl->error != OCSP_WANT_READ
     #endif
     ) {
-        ShrinkInputBuffer(ssl, NO_FORCED_FREE);
+        if (IsEncryptionOn(ssl, 0)) {
+            int extra = ssl->keys.padSz;
+        #if defined(HAVE_ENCRYPT_THEN_MAC) && !defined(WOLFSSL_AEAD_ONLY)
+            if (ssl->options.startedETMRead)
+                extra += MacSize(ssl);
+        #endif
+            ssl->buffers.inputBuffer.idx -= extra;
```

Comment:
Is this complete?  We need checks for idx >= extra.  What if idx -= extra allows size to just go down enough for static buffer use but then a couple lines later we in crease += extra that's bigger than the static buffer size.  How thoroughly has this been tested?

Reply:
I have only tested in using the case provided in ZD 10911. I will address your feedback now and push changes.

Reply:
I agree that the check to make sure extra isn't greater than the index is needed, though.